src/rust/types/src/lib.rs

```diff
@@ -0,0 +1,133 @@
+use bigdecimal::{BigDecimal, ToPrimitive};
+use chrono::{DateTime, Utc};
+use error::TypeError;
+use serde::{Deserialize, Serialize};
+
+pub mod error;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Coin {
+    pub account_address: String,
+    pub module_name: String,
+    pub struct_name: String,
+    pub symbol: String,
+    pub name: String,
+    pub decimals: i16,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Market {
+    pub market_id: u64,
+    pub base: Coin,
+    pub base_name_generic: Option<String>,
+    pub quote: Coin,
+    pub lot_size: u64,
+    pub tick_size: u64,
+    pub min_size: u64,
+    pub underwriter_id: u64,
+    pub created_at: DateTime<Utc>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct QueryMarket {
+    pub market_id: BigDecimal,
+    pub base_symbol: Option<String>,
+    pub base_name: Option<String>,
+    pub base_decimals: Option<i16>,
+    pub base_account_address: String,
+    pub base_module_name: String,
+    pub base_struct_name: String,
+    pub quote_symbol: Option<String>,
```

Comment:
Should never be none since quote asset is always a coin

Reply:
This is just out of necessity. Basically `MarketRegistrationEvent` only has `TypeInfo`, but the rest of this information has to be retrieved from `CoinInfo`.

I thought it would be better just to have the indexer put nulls in Postgres and have some other task retrieve this information upon a new insert in the markets table. This would require that these columns not be `not null`, and that in turn requires the type here to be `Option<String>`.

Reply:
Still thinking about what the best way to do this would be tbh